internal/processor/filterset/factory/config.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package factory
```

Comment:
I am not sure I understand why the data types in this file are placed in the `factory` package. They appear to be a better fit in the `filterset` package, because they describe the config of the filterset not of the factory.

Reply:
I think it may be simpler to eliminate the `factory` package and the `Factory` struct and move `CreateFilterSet` to `filterset` package as a free-standing function. I believe that will reduce the linecount without making the code less readable or usable.

Reply:
Good point! It is pretty annoying to have to import both `filterset` and `factory`. I took your suggestion and collapsed everything in `factory` into `filterset` and got rid of the Factory struct.

I originally created the `factory` package because I had a circular import issue going on between `filterset` and `regexp` packages due to `regexp.NewRegexpFilterSet` returning a `filterset.FilterSet`. But fortunately in go I can just have the regexp constructor return `*regexpFilterSet` instead of taking a dependency on `filterset` and the receiving end can still use it as a `filterset.FilterSet`.